spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -87,6 +89,53 @@ public void testRewriteDataFilesInEmptyTable() {
     assertEquals("Procedure output must match", ImmutableList.of(row(0, 0, 0L, 0)), output);
   }
 
+  @Test
+  public void testRewriteWriteOnBranch() throws Exception {
+    createPartitionTable();
+    // create 5 files for each partition (c2 = 'foo' and c2 = 'bar')
+    insertData(10);
+    Table table = Spark3Util.loadIcebergTable(spark, tableName);
+    String branch = "op_audit";
+    table.manageSnapshots().createBranch(branch).commit();
+    table.refresh();
+    spark.sql(
+        String.format("ALTER TABLE %s SET TBLPROPERTIES ('write.wap.enabled'='true')", tableName));
+    spark.sql(String.format("SET spark.wap.branch = %s", branch));
+    long lastId = table.currentSnapshot().snapshotId();
+    List<Object[]> output =
+        sql("CALL %s.system.rewrite_data_files(table => '%s')", catalogName, tableIdent);
+    table.refresh();
+    assertThat(table.currentSnapshot().snapshotId())
+        .as("rewrite should happen on branch")
+        .isEqualTo(lastId);
+  }
+
+  @Test
+  public void testRewriteReadFromBranch() throws Exception {
```

Comment:
the tests are a bit confusing. Where's the clear distinction between writing/reading to/from a branch as the naming indicates?

Reply:
not really, I started to think rewrite could read from other branch and then write here. that's where the two functions come from. but now they are testing same thing.

Reply:
and that's what makes the test confusing

Reply:
fixed